astropy/coordinates/builtin_frames/gcrs.py

```diff
@@ -32,14 +33,14 @@ class GCRS(BaseCoordinateFrame):
         The time at which the observation is taken.  Used for determining the
         position of the Earth.
     * ``obsgeoloc``
-        Vector giving the position of the observer relative to the
-        center-of-mass of the Earth, oriented the same as BCRS/ICRS.
-        The leading dimension of this vector needs to be 3.
+        The position of the observer relative to the center-of-mass of the Earth,
+        oriented the same as BCRS/ICRS.
+        Either 0, a CartesianRepresentation, or a Quantity of shape (3, ...)
```

Comment:
Not super-important, but we've tended to give fully linkable entries to astropy classes in the docstrings, i.e., something like
```
        The position of the observer relative to the center-of-mass of the Earth,
        oriented the same as BCRS/ICRS. Either 0, `~astropy.coordinates.CartesianRepresentation`,
        or proper input for one, i.e., a `~astropy.units.Quantity` with shape (3, ...) and length units.
```